backend/tracim_backend/lib/search/es_models.py

```diff
@@ -0,0 +1,142 @@
+from fnmatch import fnmatch
+import os
+import typing
+
+from elasticsearch_dsl import Boolean
+from elasticsearch_dsl import Date
+from elasticsearch_dsl import Document
+from elasticsearch_dsl import InnerDoc
+from elasticsearch_dsl import Integer
+from elasticsearch_dsl import Keyword
+from elasticsearch_dsl import Nested
+from elasticsearch_dsl import Object
+from elasticsearch_dsl import Text
+from elasticsearch_dsl import analysis
+from elasticsearch_dsl import analyzer
+
+# INFO - G.M - 2019-05-31 - Analyzer/indexing explained:
+# Instead of relying of wildcard for autocompletion which is costly and make some feature doesn't
+# work correctly, for example ranking, We use ngram mecanism.
+# This means that for work "elephant", we will index thing like "ele", "elep", "lepha", etc...
+# As we don't want to have a *text* matching but only an autocomplete matching like text*, we use
+# edge_ngram, we will only index for "elephant": "ele" , "elep" , "eleph" , etc..
+# We want that ele match elephant result, but we do not want that elephant match ele result,
+# that's why we set different analyzer for search (we search word given) and indexing (we index ngram
+# of label of content to allow autocompletion)
+
+
+edge_ngram_token_filter = analysis.token_filter(
+    "edge_ngram_filter", type="edge_ngram", min_ngram=2, max_gram=20
+)
+# INFO - G.M - 2019-05-23 - search_analyser: do search for content given an some similar word
+folding = analyzer("folding", tokenizer="standard", filter=["lowercase", "asciifolding"])
+# INFO - G.M - 2019-05-23 -  index_analysers, index edge ngram for autocompletion and strip html for indexing
+edge_ngram_folding = analyzer(
+    "edge_ngram_folding",
+    tokenizer="standard",
+    filter=["lowercase", "asciifolding", edge_ngram_token_filter],
+)
+html_folding = analyzer(
+    "html_folding",
+    tokenizer="standard",
+    filter=["lowercase", "asciifolding", edge_ngram_token_filter],
+    char_filter="html_strip",
+)
+
+DEFAULT_INDEX_DOCUMENTS_ALIAS = "tracim_contents"
```

Comment:
I feel like having a default index name in the source code is a security issue because it make software work even if index is not explicitly defined by the user (with a hardcoded value)

Reply:
fixed by making parameter mandatory